crates/js-sys/src/lib.rs

```diff
@@ -4742,3 +4787,70 @@ pub fn global() -> Object {
 
     GLOBAL.with(|g| g.clone())
 }
+
+macro_rules! arrays {
+    ($($name:ident: $ty:ident,)*) => ($(
+        impl $name {
+            /// Creates a JS typed array which is a few into wasm's linear
+            /// memory at the slice specified.
+            ///
+            /// This function returns a new typed array which is a view into
+            /// wasm's memory. This view does not copy the underlying data.
+            ///
+            /// # Unsafety
+            ///
+            /// Views into WebAssembly memory are only valid so long as the
+            /// backing buffer isn't resized in JS. Once this function is called
+            /// any future calls to `Box::new` (or malloc of any form) may cause
+            /// the returned value here to be invalidated. Use with caution!
+            ///
+            /// Additionally the returned object can be safely mutated but the
+            /// input slice isn't guaranteed to be mutable.
+            ///
+            /// Finally, the returned objet is disconnected from the input
+            /// slice's lifetime, so there's no guarantee that the data is read
+            /// at the right time.
+            pub unsafe fn view(rust: &[$ty]) -> $name {
+                let buf = wasm_bindgen::memory();
+                let mem = buf.unchecked_ref::<WebAssembly::Memory>();
+                $name::new_with_byte_offset_and_length(
+                    &mem.buffer(),
+                    rust.as_ptr() as u32,
+                    rust.len() as u32,
+                )
+            }
+
+            /// Copy the contents of this JS typed array into the destination
+            /// Rust slice.
+            ///
+            /// This function will efficiently copy the memory from a typed
```

Comment:
Curious, what does "efficiently" mean here?

Thanks!

Reply:
Ah this just means that it uses the `set` API which eventually translates to something like a raw `memcpy`. That's the fastest way (even in Rust) to copy chunks of memory. 

The slow way would be to copy bytes one-by-one in wasm which misses on on things like vectorization and eliding bounds checks

Reply:
_Gains knowledge_

Cool - thanks!!